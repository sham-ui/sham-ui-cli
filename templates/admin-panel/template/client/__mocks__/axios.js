const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
