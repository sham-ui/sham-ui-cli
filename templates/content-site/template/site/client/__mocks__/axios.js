const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
