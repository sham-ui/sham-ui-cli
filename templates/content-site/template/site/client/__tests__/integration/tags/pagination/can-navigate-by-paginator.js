import setupUnsafe from 'sham-ui-unsafe';
import axios from 'axios';
import setup, { app } from '../../helpers';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
} );

it( 'go to page', async() => {
    expect.assertions( 4 );

    const DI = setup();
    setupUnsafe( DI );

    axios
        .useDefaultMocks()
        .use( 'get', '/articles', {
            articles: [
                {
                    'title': 'Первая тестовая статья',
                    'slug': 'pervya-testovaya-statya',
                    'category': {
                        'name': 'Первая',
                        'slug': 'pervya'
                    },
                    'content': 'Short content',
                    'createdAt': '2022-05-24 19:54:20 +0000 UTC'
                }
            ],
            meta: {
                total: 100,
                limit: 9,
                tag: 'Быт'
            }
        } )
    ;

    history.pushState( {}, '', 'http://client.example.com/tag/byt/5' );

    await app.start( DI );

    app.click( '[data-test-page="6"]' );
    await app.waitRendering();

    app.checkBody();
    expect( axios.mocks.get ).toHaveBeenCalledTimes( 2 );
    expect( axios.mocks.get.mock.calls[ 1 ] ).toEqual( [
        '/articles',
        { 'limit': 9, 'offset': 45, 'tag': 'byt' }
    ] );
    expect( window.location.href ).toBe( 'http://client.example.com/tag/byt/6/' );
} );

it( 'go to page (ssr & rehydrate)', async() => {
    expect.assertions( 7 );

    axios
        .useDefaultMocks()
        .use( 'get', '/articles', {
            articles: [
                {
                    'title': 'Первая тестовая статья',
                    'slug': 'pervya-testovaya-statya',
                    'category': {
                        'name': 'Первая',
                        'slug': 'pervya'
                    },
                    'content': 'Short content',
                    'createdAt': '2022-05-24 19:54:20 +0000 UTC'
                }
            ],
            meta: {
                total: 100,
                limit: 9,
                tag: 'Быт'
            }
        } )
    ;

    await app.ssrAndRehydrate( 'http://client.example.com/tag/byt/5' );

    expect( window.location.href ).toBe( 'http://client.example.com/tag/byt/5' );
    expect( axios.mocks.get ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.get.mock.calls[ 0 ] ).toEqual( [
        '/articles',
        { 'limit': 9, 'offset': 36, 'tag': 'byt' }
    ] );

    app.click( '[data-test-page="6"]' );
    await app.waitRendering();

    app.checkBody();
    expect( axios.mocks.get ).toHaveBeenCalledTimes( 2 );
    expect( axios.mocks.get.mock.calls[ 1 ] ).toEqual( [
        '/articles',
        { 'limit': 9, 'offset': 45, 'tag': 'byt' }
    ] );
    expect( window.location.href ).toBe( 'http://client.example.com/tag/byt/6/' );
} );
