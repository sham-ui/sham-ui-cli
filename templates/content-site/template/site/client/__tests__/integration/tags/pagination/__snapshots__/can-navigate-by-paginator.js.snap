// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`go to page (ssr & rehydrate) 1`] = `
"<nav class=\\"navbar navbar-expand-lg fixed-top\\">
  <div class=\\"container-fluid\\">
    <div class=\\"logo\\">
      <a href=\\"\\">
        <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
      </a>
    </div>
    <!--0-->
    <div class=\\"collapse navbar-collapse\\">
      <ul class=\\"navbar-nav ml-auto mr-auto\\">
        <li class=\\"nav-item\\">
          <a href=\\"\\" class=\\"nav-link \\">Home</a>
        </li>
        <li class=\\"nav-item\\">
          <a href=\\"/contact/\\" class=\\"nav-link\\">Contact</a>
        </li>
      </ul>
    </div>
    <div class=\\"navbar-right ml-auto\\">
      <div class=\\"theme-switch-wrapper\\">
        <label for=\\"checkbox\\" class=\\"theme-switch\\">
          <input type=\\"checkbox\\" id=\\"checkbox\\">
          <div class=\\"slider round\\"></div>
        </label>
      </div>
      <!--1-->
      <div class=\\"search-icon\\">
        <i class=\\"icon-search\\"></i>
      </div>
      <button type=\\"button\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    </div>
  </div>
</nav>
<!--0-->
<div data-test-page-body=\\"\\">
  <section class=\\"section pt-55\\">
    <div class=\\"container-fluid\\">
      <div class=\\"row\\">
        <div class=\\"categorie-title\\">
          <h3>Tag : <span>Быт</span></h3>
        </div>
      </div>
    </div>
    <!--0-->
    <!--1-->
    <div class=\\"container-fluid\\">
      <div class=\\"row\\">
        <div class=\\"col-lg-4 col-md-6\\">
          <div class=\\"post-card\\">
            <div class=\\"post-card-content\\">
              <a class=\\"categorie\\" href=\\"/category/pervya/1/\\">Первая</a>
              <h5>
                <a href=\\"/pervya-testovaya-statya\\">Первая тестовая статья</a>
              </h5>
              <p>Short content</p>
              <div class=\\"post-card-info\\">
                <ul class=\\"list-inline\\">
                  <li>среда, 25 мая 2022 г., 02:54:20</li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <!--0-->
        <!--2-->
        <div class=\\"col-lg-12\\">
          <div class=\\"pagination mt--10\\">
            <ul class=\\"list-inline\\">
              <!--0-->
              <li>
                <a data-test-page=\\"1\\" class=\\"\\" href=\\"/tag/byt/1/\\">1</a>
              </li>
              <li class=\\"mr-2\\">...</li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"4\\" class=\\"\\" href=\\"/tag/byt/4/\\">4</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"5\\" class=\\"\\" href=\\"/tag/byt/5/\\">5</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"6\\" class=\\"active\\" href=\\"/tag/byt/6/\\">6</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"7\\" class=\\"\\" href=\\"/tag/byt/7/\\">7</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"8\\" class=\\"\\" href=\\"/tag/byt/8/\\">8</a>
              </li>
              <!--1-->
              <li class=\\"mr-2\\">...</li>
              <!--0-->
              <li>
                <a data-test-page=\\"12\\" class=\\"\\" href=\\"/tag/byt/12/\\">12</a>
              </li>
              <!--1-->
            </ul>
          </div>
        </div>
      </div>
    </div>
  </section>
  <!--0-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->
<!--1-->
<footer class=\\"footer\\">
  <div class=\\"container-fluid\\">
    <div class=\\"row\\">
      <div class=\\"col-lg-12\\">
        <div class=\\"back\\">
          <a href=\\"#\\" class=\\"back-top\\">
            <i class=\\"icon-up\\"></i>
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
<!--2-->
<!--3-->
<!--0-->
<!--0-->
<script>
  window.data = {
    \\"0\\": {
      \\"C\\": [\\"1\\"],
      \\"N\\": [11]
    },
    \\"1\\": {
      \\"C\\": [\\"2\\", \\"6\\", \\"7\\"],
      \\"N\\": [1, 7, 9, 10]
    },
    \\"2\\": {
      \\"C\\": [\\"3\\", \\"4\\", \\"5\\"],
      \\"N\\": [0, [0, 0],
        [0, 0, 1],
        [0, 0, 2],
        [0, 0, 3],
        [0, 0, 3, 1],
        [0, 0, 3, 2],
        [0, 0, 3, 2, 0],
        [0, 0, 3, 3],
        [0, 0, 3, 3, 0]
      ]
    },
    \\"3\\": {
      \\"N\\": [0, [0, 0],
        [0, 0, 0]
      ]
    },
    \\"4\\": {
      \\"N\\": [0, [0, 0],
        [0, 0, 0],
        [0, 0, 0, 0],
        [0, 1],
        [0, 1, 0],
        [0, 1, 0, 0]
      ]
    },
    \\"5\\": {
      \\"N\\": [0, [0, 0],
        [0, 0, 0],
        [0, 0, 1]
      ]
    },
    \\"6\\": {
      \\"N\\": [8, [8, 0],
        [8, 0, 0],
        [8, 0, 0, 0],
        [8, 0, 0, 0, 0],
        [8, 0, 0, 0, 0, 0],
        [8, 0, 0, 0, 0, 0, 0]
      ]
    },
    \\"7\\": {
      \\"C\\": [\\"8\\"],
      \\"N\\": [6]
    },
    \\"8\\": {
      \\"C\\": [\\"9\\"]
    },
    \\"9\\": {
      \\"O\\": {
        \\"tag\\": \\"Быт\\"
      },
      \\"C\\": [\\"10\\"],
      \\"N\\": [5]
    },
    \\"10\\": {
      \\"O\\": {
        \\"dataLoaded\\": true,
        \\"errors\\": [],
        \\"page\\": 5,
        \\"total\\": 100,
        \\"articles\\": [{}]
      },
      \\"C\\": [\\"11\\"],
      \\"N\\": [4]
    },
    \\"11\\": {
      \\"C\\": [\\"14\\"],
      \\"N\\": [3]
    },
    \\"14\\": {
      \\"C\\": [\\"15\\"],
      \\"N\\": [2, [2, 1]]
    },
    \\"15\\": {
      \\"C\\": [\\"16\\", \\"17\\", \\"18\\"],
      \\"N\\": [0, [0, 3],
        [0, 4],
        [0, 5],
        [0, 5, 0],
        [0, 5, 0, 2],
        [0, 5, 0, 3]
      ]
    },
    \\"16\\": {
      \\"N\\": [0, 1, [1, 0],
        [1, 0, 0],
        [1, 0, 0, 0],
        [1, 0, 0, 0, 0],
        [1, 0, 0, 0, 1],
        [1, 0, 0, 0, 1, 0], 2
      ]
    },
    \\"17\\": {
      \\"C\\": [\\"28\\"],
      \\"N\\": [1]
    },
    \\"18\\": {
      \\"C\\": [\\"19\\", \\"20\\", \\"21\\", \\"22\\", \\"23\\", \\"24\\", \\"25\\"],
      \\"N\\": [0, [0, 0]]
    },
    \\"19\\": {
      \\"C\\": [\\"26\\"],
      \\"N\\": [0, 1, [1, 0],
        [1, 0, 0], 3
      ]
    },
    \\"20\\": {
      \\"N\\": [4, 5, [5, 0],
        [5, 0, 0], 6
      ]
    },
    \\"21\\": {
      \\"N\\": [7, 8, [8, 0],
        [8, 0, 0], 9
      ]
    },
    \\"22\\": {
      \\"N\\": [10, 11, [11, 0],
        [11, 0, 0], 12
      ]
    },
    \\"23\\": {
      \\"N\\": [13, 14, [14, 0],
        [14, 0, 0], 15
      ]
    },
    \\"24\\": {
      \\"N\\": [16, 17, [17, 0],
        [17, 0, 0], 18
      ]
    },
    \\"25\\": {
      \\"C\\": [\\"27\\"],
      \\"N\\": [20, 21, [21, 0],
        [21, 0, 0], 22
      ]
    },
    \\"26\\": {
      \\"N\\": [2, [2, 0]]
    },
    \\"27\\": {
      \\"N\\": [19, [19, 0]]
    },
    \\"28\\": {
      \\"N\\": [0, [0, 0],
        [0, 0, 0],
        [0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 1],
        [0, 0, 0, 1, 0],
        [0, 0, 0, 1, 0, 0],
        [0, 0, 0, 2],
        [0, 0, 0, 2, 0],
        [0, 0, 0, 3],
        [0, 0, 0, 3, 0],
        [0, 0, 0, 3, 0, 0],
        [0, 0, 0, 3, 0, 0, 0]
      ]
    },
    \\"app\\": {
      \\"C\\": [\\"0\\"],
      \\"N\\": [12]
    }
  };
</script>
<script src=\\"/s.min.js\\"></script>
<script>
  System.import('/bundle.js');
</script>"
`;

exports[`go to page 1`] = `
"<nav class=\\"navbar navbar-expand-lg fixed-top\\">
  <div class=\\"container-fluid\\">
    <div class=\\"logo\\">
      <a href=\\"\\">
        <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
      </a>
    </div>
    <!--0-->
    <div class=\\"collapse navbar-collapse\\">
      <ul class=\\"navbar-nav ml-auto mr-auto\\">
        <li class=\\"nav-item\\">
          <a class=\\"nav-link \\" href=\\"\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
          <a class=\\"nav-link\\" href=\\"/contact/\\">Contact</a>
        </li>
      </ul>
    </div>
    <div class=\\"navbar-right ml-auto\\">
      <div class=\\"theme-switch-wrapper\\">
        <label class=\\"theme-switch\\" for=\\"checkbox\\">
          <input type=\\"checkbox\\" id=\\"checkbox\\">
          <div class=\\"slider round\\"></div>
        </label>
      </div>
      <!--1-->
      <div class=\\"search-icon\\">
        <i class=\\"icon-search\\"></i>
      </div>
      <button class=\\"navbar-toggler\\" type=\\"button\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    </div>
  </div>
</nav>
<!--0-->
<div data-test-page-body=\\"\\">
  <section class=\\"section pt-55\\">
    <div class=\\"container-fluid\\">
      <div class=\\"row\\">
        <div class=\\"categorie-title\\">
          <h3>Tag : <span>Быт</span></h3>
        </div>
      </div>
    </div>
    <!--0-->
    <!--1-->
    <div class=\\"container-fluid\\">
      <div class=\\"row\\">
        <div class=\\"col-lg-4 col-md-6\\">
          <div class=\\"post-card\\">
            <div class=\\"post-card-content\\">
              <a class=\\"categorie\\" href=\\"/category/pervya/1/\\">Первая</a>
              <h5>
                <a href=\\"/pervya-testovaya-statya\\">Первая тестовая статья</a>
              </h5>
              <p>Short content</p>
              <div class=\\"post-card-info\\">
                <ul class=\\"list-inline\\">
                  <li>среда, 25 мая 2022 г., 02:54:20</li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <!--0-->
        <!--2-->
        <div class=\\"col-lg-12\\">
          <div class=\\"pagination mt--10\\">
            <ul class=\\"list-inline\\">
              <!--0-->
              <li>
                <a data-test-page=\\"1\\" class=\\"\\" href=\\"/tag/byt/1/\\">1</a>
              </li>
              <li class=\\"mr-2\\">...</li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"4\\" class=\\"\\" href=\\"/tag/byt/4/\\">4</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"5\\" class=\\"\\" href=\\"/tag/byt/5/\\">5</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"6\\" class=\\"active\\" href=\\"/tag/byt/6/\\">6</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"7\\" class=\\"\\" href=\\"/tag/byt/7/\\">7</a>
              </li>
              <!--1-->
              <!--0-->
              <li>
                <a data-test-page=\\"8\\" class=\\"\\" href=\\"/tag/byt/8/\\">8</a>
              </li>
              <!--1-->
              <li class=\\"mr-2\\">...</li>
              <!--0-->
              <li>
                <a data-test-page=\\"12\\" class=\\"\\" href=\\"/tag/byt/12/\\">12</a>
              </li>
              <!--1-->
            </ul>
          </div>
        </div>
      </div>
    </div>
  </section>
  <!--0-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->
<!--1-->
<footer class=\\"footer\\">
  <div class=\\"container-fluid\\">
    <div class=\\"row\\">
      <div class=\\"col-lg-12\\">
        <div class=\\"back\\">
          <a href=\\"#\\" class=\\"back-top\\">
            <i class=\\"icon-up\\"></i>
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
<!--2-->
<!--3-->
<!--0-->
<!--0-->"
`;
